lib/src/HttpRequestImpl.h

```diff
@@ -705,6 +719,7 @@ class HttpRequestImpl : public HttpRequest
     RequestStreamReaderPtr streamReaderPtr_;
     std::exception_ptr streamExceptionPtr_;
     bool startProcessing_{false};
+    std::function<bool()> checkConnCallback_;
```

Comment:
I prefer to hold a weak_ptr of the connection.

Reply:
fixed